src/a4/domain/JailSpace.java

```diff
@@ -0,0 +1,22 @@
+package a4.domain;
+
+import java.util.HashMap;
+
+public class JailSpace extends BoardSpace {
+	public JailSpace() {
+	}
+
+	HashMap<Player, Integer> playersInJail = new HashMap<Player, Integer>();
+
+	void getOutOfJail(Player player_leaving_jail) {
+		playersInJail.remove(player_leaving_jail);
+	}
+
+	void putPlayerInJail(Player player_to_put_in_jail) {
+		playersInJail.put(player_to_put_in_jail, 0);
+	}
+
+	int getAttempts(Player player_in_jail) {
+		return playersInJail.get(player_in_jail);
+	}
```

Comment:
Do we need a method to increment the number of times the player has attempted to get out of jail?